### ISSUE TITLE
@now/next@2.3.2 builder breaks routing with now dev

### ISSUE THREAD

**Issue by andrewdavidkay** [vercel/vercel#3486]
I am unable to navigate to dynamic routes via full page refresh. Also static assets 404. Downgrading to `@now/next@2.3.1` fixes things for me. I am using a monorepo setup. All code is open source here: https://github.com/iamandyk/fitfam. Here is my `now.json`

```
{
  "version": 2,
  "name": "FitFam",
  "alias": "fitfam.me",
  "env": {
    "API_URL": "@fitfam-api-url",
    "GRAPHQL_API_URL": "@fitfam-graphql-api-url",
    "MYSQL_URL": "@fitfam-mysql-url",
    "FACEBOOK_APP_ID": "@fitfam-facebook-app-id",
    "FACEBOOK_APP_SECRET": "@fitfam-facebook-app-secret",
    "JWT_SECRET": "@fitfam-jwt-secret"
  },
  "build": {
    "env": {
      "MYSQL_URL": "@fitfam-mysql-url",
      "API_URL": "@fitfam-api-url",
      "GRAPHQL_API_URL": "@fitfam-graphql-api-url"
    }
  },
  "builds": [
    {
      "src": "api/routes/*.ts",
      "use": "@now/node"
    },
    {
      "src": "api/src/index.ts",
      "use": "@now/node",
      "config": {
        "maxLambdaSize": "25mb"
      }
    },

    { "src": "www/package.json", "use": "@now/next@2.3.1" }
  ],
  "routes": [
    {
      "src": "/api/facebook-auth",
      "dest": "/api/routes/facebook-auth.ts"
    },
    {
      "src": "/api/(.*)",
      "dest": "/api/src/index.ts"
    },
    { "src": "/(.*)", "dest": "/www/$1", "continue": true }
  ]
}
```

**Comment by ivanhueso** [comment]
The same happen to me.

**Comment by rosscampbell** [comment]
Ditto.

**Comment by lipp** [comment]
Also having problems with 2.3.2 . With `now dev` routing seems broken, e.g. assets from the `/public` folder can't be found. Production deployment works normal.

**Comment by jagaapple** [comment]
I have the same problems too.
Regardless whether or not a project has configurations for Next.js and Now, dynamic routing seems be broken. I've deleted caches of built files and `node_modules` then install again, but it doesn't work fine yet.

For example, I put files like the following firstly.
```
/sample
  - /pages
    - /users
      - [id].jsx
    - index.jsx
  - package-lock.json
  - package.json
```

When I go to pages using dynamic routing such as `/users/123` , the pages raise errors like the following.

![image](https://user-images.githubusercontent.com/20621562/71661756-95356100-2d92-11ea-821e-a2417c5f8b46.png)

If I go to these pages by only History API (e.g. `<Link />` and `Router.push()` ), it's successful to render them. Also I can access pages not using dynamic routing such as `/` (or `/index).

**Comment by josantana** [comment]
Same here. Downgrading to `@now/next@2.3.1` "fixed" my application, as @iamandyk mentioned. Something related with dynamic routes, for sure.

**Comment by dnnsk** [comment]
Same story. public folder throws FILE_NOT_FOUND as well as dynamic routes. Please, fix.

**Comment by timswalling** [comment]
Same for me. Note that you need to set `@now/next@2.3.1` via the `builds` section of `now.json` (eg `"builds": [{ "src": "package.json", "use": "@now/next@2.3.1" }],`) and not via NPM/Yarn.

**Comment by qonl** [comment]
Same issue. Downgrading to @now/next@2.3.1 fixed the issue for me.

**Comment by wildtrek** [comment]
Same issue here; Downgrading to @now/next@2.3.1 also fix it for me.

**Comment by zxkxyz** [comment]
Same issue with Now v17.1.1, @timswalling's solution fixed the issue for now...

**Comment by schwamic** [comment]
same here with `@now/next@2.5.1`; works with:

```json
"version": 2,
"builds": [
  {
    "src": "package.json",
    "use": "@now/next@2.4.0"
  }
]
```

**Comment by jagaapple** [comment]
I just confirmed it works fine in v18.0.0 🎉

**Comment by mcsdevv** [comment]
Closing this at it has been resolved from 18.0.0 onwards.
